docs/protocol.md

```diff
@@ -662,7 +673,235 @@ Response (in case of success)
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "message_ids_by_channel_id": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                            "channel": {
+                                "description": "[String] name of the channel",
+                                "type": "string",
+                                "pattern": "^/root(/[^/]+)*$"
+                            },
+                            "message_ids": {
+                                "description": "[Array] of message_ids corresponding to that channel",
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "required": ["channel", "message_ids"]
+                    }
+                }
+            },
+            "required": ["message_ids_by_channel_id"]
+        },
+        "jsonrpc": {
+            "$comment": "Defined by the parent, but needed here for the validation"
+        },
+
+        "id": {
+            "type": "integer"
+        }
+    },
+    "required": ["method", "params", "id", "jsonrpc"]
+}
+
+```
+</details>
+
+## Retrieving messages from server using ids 
+
+🧭 **RPC Message** > **Query** > **Get messages by their ids**
+  
+The purpose of this RPC is to request missed messages so far.
+
+This call is generally triggered upon the reception of a hearbeat message, in case the server hasn't already received some of the messages contained in such heartbeat but it could also used in other contexts. 
+
+RPC 
+
+```json5
+// ../protocol/examples/query/get_messages_by_id/get_messages_by_id.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 6,
+    "method": "get_messages_by_id",
+    "params": {
+        "message_ids_by_channel_id": [
+            {
+                "channel": "/root/nLghr9_P406lfkMjaNWqyohLxOiGlQee8zad4qAfj18=/social/8qlv4aUT5-tBodKp4RszY284CFYVaoDZK6XKiw9isSw=",
+                "message_ids": ["DCBX48EuNO6q-Sr42ONqsj7opKiNeXyRzrjqTbZ_aMI="]
+            },
+            {
+                "channel": "/root/nLghr9_P406lfkMjaNWqyohLxOiGlQee8zad4qAfj18=/HnXDyvSSron676Icmvcjk5zXvGLkPJ1fVOaWOxItzBE=",
+                "message_ids": [
+                    "z6SbjJ0Hw36k8L09-GVRq4PNmi06yQX4e8aZRSbUDwc=",
+                    "txbTmVMwCDkZdoaAiEYfAKozVizZzkeMkeOlzq5qMlg="
+                ]
+            }
+        ]
+    }
+}
+
+```
+
+Response (in case of success)
+
+```json5
+// ../protocol/examples/answer/get_messages_by_id_ans.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 6,
+    "result": {
+        "messages_by_channel_id": [
```

Comment:
did you have something specific in mind when you added an object with a single field (messages_by_channel_id) wrapping the actual result ?

Reply:
I think the problem was that `array` was not an accepted type but `object` was. But since we are using a map now this is no longer a problem.